parsel/xpathfuncs.py

```diff
@@ -0,0 +1,33 @@
+from lxml import etree
+from six import iteritems, get_function_code
+
+_XPATH_FUNCS = {}
+
+
+def register(func):
+    fname = get_function_code(func).co_name.replace('_', '-')
```

Comment:
I realize the decorator idea might have been sparkled by my own example code... :grin: 
But I hadn't thought through the problems of making the decorator available in the library and having to support all the weird edge cases that people might try to use (methods, etc). =)